WireUI/Tests/WireMoveToFolderUITests/FolderPickerViewModelTests.swift

```diff
@@ -0,0 +1,138 @@
+        mockSelectionUseCase = nil
+        super.tearDown()
+    }
+
+    // MARK: - Initialization
+
+    func test_init_loadsFoldersFromDirectory() {
+        // GIVEN
+        let folders = [
+            Folder(identifier: UUID(), name: "Work"),
+            Folder(identifier: UUID(), name: "Personal")
+        ]
+        mockDirectory.allFolders = folders
+
+        // WHEN
+        createSUT()
+
+        // THEN
+        XCTAssertEqual(sut.folders, folders)
+    }
+
+    // MARK: - Folder Selection
+
+    func test_select_invokesUseCase() {
+        // GIVEN
+        let folder = Folder(identifier: UUID(), name: "Work")
+        let conversation = Conversation(identifier: UUID(), currentFolderIdentifier: nil)
+        mockSelectionUseCase.invokeFolderConversation_MockMethod = { _, _ in }
+        createSUT(conversation: conversation)
+
+        // WHEN
+        sut.select(folder)
+
+        // THEN
+        XCTAssertEqual(mockSelectionUseCase.invokeFolderConversation_Invocations.count, 1)
+        XCTAssertEqual(mockSelectionUseCase.invokeFolderConversation_Invocations.first?.folder, folder)
+    }
+
+    // MARK: - Selection State
+
+    func test_isSelected_returnsTrue_whenFolderMatchesCurrentFolder() {
+        // GIVEN
+        let folderID = UUID()
+        let conversation = Conversation(identifier: UUID(), currentFolderIdentifier: folderID)
+        let folder = Folder(identifier: folderID, name: "Work")
+        createSUT(conversation: conversation)
+
+        // WHEN
+        let isSelected = sut.isSelected(folder)
+
+        // THEN
+        XCTAssertTrue(isSelected)
+    }
+
+    func test_isSelected_returnsFalse_whenFolderDoesNotMatchCurrentFolder() {
+        // GIVEN
+        let conversation = Conversation(identifier: UUID(), currentFolderIdentifier: UUID())
+        let folder = Folder(identifier: UUID(), name: "Work")
+        createSUT(conversation: conversation)
+
+        // WHEN
+        let isSelected = sut.isSelected(folder)
+
+        // THEN
+        XCTAssertFalse(isSelected)
+    }
+
+    func test_isSelected_returnsFalse_whenFolderIdentifierIsNil() {
+        // GIVEN
+        let conversation = Conversation(identifier: UUID(), currentFolderIdentifier: UUID())
+        let folder = Folder(identifier: nil, name: "Work")
+        createSUT(conversation: conversation)
+
+        // WHEN
+        let isSelected = sut.isSelected(folder)
+
+        // THEN
+        XCTAssertFalse(isSelected)
+    }
+
+    // MARK: - Helpers
+
+    private func createSUT(
+        conversation: Conversation = Conversation(identifier: UUID(), currentFolderIdentifier: nil),
+        directory: MockFolderDirectoryType? = nil,
+        selectionUseCase: MockMoveConversationToFolderUseCaseType? = nil
+    ) {
+        sut = FolderPickerViewModel(
+            conversation: conversation,
+            directory: directory ?? mockDirectory,
+            selectionUseCase: selectionUseCase ?? mockSelectionUseCase
+        )
+    }
```

Comment:
In my opinion this code should be deleted.
Having default arguments is error prone and in this case the arguments are never set.